### PR TITLE
Ensure that minified asset detection is SSL-friendly.

### PR DESCRIPTION
**Pull request by boonebgorges**
I ran into a problem with The Events Calendar where JS assets were not loading on post-new.php?post_type=tribe_events. After a great deal of debugging, I traced it down to the problem that was previously quite thoroughly explained in this wordpress.org ticket by @earnjam : https://wordpress.org/support/topic/use-of-wp_content_url-breaks-plugin-on-admin-ssl-only-sites/ As he notes there, the problem arises on `FORCE_SSL_ADMIN`, but it's not only there: any time `get_option( 'home' )` is http:// but the current URL is https://, the problem arises. (SSL-optional sites, server-level rewrites/redirects, etc.)

Basically, when performing string comparisons on URL, you must ensure that your needle shares the scheme with the current haystack URL. There's a number of ways to do this. For greatest consistency and compatibility with mu-plugins, I chose to run everything through `set_url_scheme()` in this PR.

If for some reason you choose not to implement something like this, please provide a filter that I can use to work around the problem without forking the plugin :-)

### CODE REVIEW

**Comment by barryhughes** [comment]
Thanks @boonebgorges - thanks for the PR!

What do you think of the change [made here](https://github.com/moderntribe/tribe-common/blob/release/M17.16/src/Tribe/Assets.php#L205-L268) (the release/M17.16 branch is going to be the basis of our next release), which were put in place for similar reasons?

As a related aside, a filter could be useful here even so.

**Comment by boonebgorges** [comment]
Oh! https://github.com/moderntribe/tribe-common/commit/3d7aedf8c198b5e623225a996e9078954c11ffd2 is the same as this PR. Sorry for not seeing this - I didn't know how your branch system worked.

**Comment by barryhughes** [comment]
No problem - we'll have a think about ways to make that a little clearer (we recently re-structured the way we use branches).

As a heads-up, though I can't guarantee anything, we _should_ see this change ship next week.

Thanks again!

**Comment by boonebgorges** [comment]
Many thanks for the quick follow-up!
